sql/core/src/main/scala/org/apache/spark/sql/api/java/Row.scala

```diff
@@ -125,6 +126,7 @@ object Row {
     case seq: scala.collection.Seq[_] =>
       JavaConversions.seqAsJavaList(seq.map(toJavaValue))
     case decimal: BigDecimal => decimal.underlying()
+    case date: Date => date.toJavaDate
```

Comment:
That's true, now it is a different story, from what it was when I wrote these codes...